### PR TITLE
Feature/account service

### PR DESCRIPTION
**Pull request by ja573**
Replaced CookieService with AccountService, moving all logic around tokens to this single place. Token is now stored in local storage instead of a cookie as stdweb wasn't able to set a cookie using the js macro anymore

### CODE REVIEW

File: thoth-app/src/component/navbar.rs [hunk 1]
```diff
@@ -5,36 +5,60 @@ use yew_router::prelude::*;
...
+                Msg::Logout()
+            }),
+            false => self.link.callback(|e: MouseEvent| {
+                e.prevent_default();
+                Msg::Login()
+            }),
         };
-        let auth_button = match authenticated {
+        let auth_button = match self.account_service.is_loggedin() {
             true => "Logout",
```
[the remark this answers is not in the document]
Comment (ja573) [reply thoth-app/src/component/navbar.rs:62]: Yes to this and the comment below. In this PR I'm just replacing what is currently in place. The next step is to use the accountservice at the root component and have all child components inherit its state through properties, which should trigger a global component update if the account changes

File: thoth-app/src/service/account.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+impl AccountService {
+    pub fn new() -> Self {
+        let login_route = Route::from(AppRoute::Login);
+        Self { login_route }
+    }
+
+    pub fn get_token(&self) -> Option<String> {
+        let storage_service = StorageService::new(Area::Local).expect(STORAGE_ERROR);
+        if let Ok(token) = storage_service.restore(SESSION_KEY) {
+            log::debug!("Get token: {}", token);
```
[the remark this answers is not in the document]
Comment (ja573) [reply thoth-app/src/service/account.rs:23]: You're right, these need to go. I had confused it with the API logging, assuming they would only be set if `RUST_BACKTRACE=1` is too, but it doesn't work that way in the GUI

File: thoth-app/src/service/account.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+use yew_router::route::Route;
+
+use crate::route::AppRoute;
+use crate::SESSION_KEY;
+
+pub struct AccountService {
+    login_route: Route,
+}
+
+const STORAGE_ERROR: &str = "local storage is disabled";
```
[the remark this answers is not in the document]
Comment (ja573) [reply thoth-app/src/service/account.rs]: This one was fixed with fbfb798

File: thoth-app/src/service/account.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ja573) [reply thoth-app/src/service/account.rs]: I've removed these now.

**Review by rhigman — APPROVED** [review]

**Pull request merged**